Task 1: Loading the Data:

*  Load the Titanic dataset into a Pandas DataFrame.
*  Dataset: Titanic Dataset
*  You can download dataset from here

In [ ]:
# Import pandas
import pandas as pd

# Get the csv from github
url = "https://raw.githubusercontent.com/angseesiang/lithan/main/titanic.csv"

# Read the csv file
titanic = pd.read_csv(url, index_col=0)
print(titanic)

             Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   
...               ...     ...   
887                 0       2   
888                 1       1   
889                 0       3   
890                 1       1   
891                 0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   
...       

Task 2: Initial Exploration:

*  Display the first 5 rows of the dataset to get an overview.
*  Provide basic information about the dataset using the info() method.

In [ ]:
# Print the first 5 rows
print(titanic.head().info())

<class 'pandas.core.frame.DataFrame'>
Index: 5 entries, 1 to 5
Data columns (total 11 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  5 non-null      int64  
 1   Pclass    5 non-null      int64  
 2   Name      5 non-null      object 
 3   Sex       5 non-null      object 
 4   Age       5 non-null      float64
 5   SibSp     5 non-null      int64  
 6   Parch     5 non-null      int64  
 7   Ticket    5 non-null      object 
 8   Fare      5 non-null      float64
 9   Cabin     2 non-null      object 
 10  Embarked  5 non-null      object 
dtypes: float64(2), int64(4), object(5)
memory usage: 480.0+ bytes
None


Task 3: Handling Missing Values:

*  Identify and count the number of missing values in each column.
*  Decide on an appropriate strategy for handling missing values in at least two columns and implement it.

In [ ]:
# Display missing values in each column
missing_values = titanic.isnull().sum()
print("Missing values per column:\n", missing_values)

Missing values per column:
 Survived      0
Pclass        0
Name          0
Sex           0
Age         177
SibSp         0
Parch         0
Ticket        0
Fare          0
Cabin       687
Embarked      2
dtype: int64


Task 4: Descriptive Statistics:

*  Generate descriptive statistics for the numerical columns using the describe() method.

In [ ]:
# Generate descriptive statistics for numerical columns
numeric_stats = titanic.describe()
print(numeric_stats)

         Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200


Task 5: Unique Values:

*  List unique values in the 'Pclass' column.
*  Count the occurrences of each unique value in the 'Embarked' column.

In [ ]:
# List unique values in 'Pclass'
unique_pclass = titanic['Pclass'].unique()
print("Unique values in 'Pclass':", unique_pclass)

Unique values in 'Pclass': [3 1 2]


Task 6: Data Manipulation:

*  Create a new column named 'FamilySize' that represents the sum of 'SibSp' and 'Parch'.
*  Drop the 'Cabin' column from the dataset.

In [ ]:
# Create 'FamilySize' by summing 'SibSp' and 'Parch'
titanic['FamilySize'] = titanic['SibSp'] + titanic['Parch']

# Drop the 'Cabin' column
titanic.drop('Cabin', axis=1, inplace=True)

# Show updated DataFrame columns
print(titanic.columns)

# Preview the first few rows
print(titanic.head())

Index(['Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket',
       'Fare', 'Embarked', 'FamilySize'],
      dtype='object')
             Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Embarked  F

Task 7: Filtering and Sorting:

*  Create a new DataFrame containing only passengers with an age greater than 30.
*  Sort the DataFrame by 'Fare' in descending order.

In [ ]:
# Filter passengers with Age > 30
age_above_30 = titanic[titanic['Age'] > 30]

# Sort by 'Fare' in descending order
age_above_30_sorted = age_above_30.sort_values(by='Fare', ascending=False)

# Display the first few rows
print(age_above_30_sorted.head())

             Survived  Pclass  \
PassengerId                     
680                 1       1   
738                 1       1   
259                 1       1   
439                 0       1   
300                 1       1   

                                                        Name     Sex   Age  \
PassengerId                                                                  
680                       Cardeza, Mr. Thomas Drake Martinez    male  36.0   
738                                   Lesurer, Mr. Gustave J    male  35.0   
259                                         Ward, Miss. Anna  female  35.0   
439                                        Fortune, Mr. Mark    male  64.0   
300          Baxter, Mrs. James (Helene DeLaudeniere Chaput)  female  50.0   

             SibSp  Parch    Ticket      Fare Embarked  FamilySize  
PassengerId                                                         
680              0      1  PC 17755  512.3292        C           1  
738           

Task 8: Data Analysis:

*  Calculate the survival rate for different passenger classes ('Pclass').
*  Identify and display the passenger with the highest 'Fare'.

In [ ]:
# Calculate survival rate by 'Pclass'
survival_rate_by_class = titanic.groupby('Pclass')['Survived'].mean()
print("Survival rate by passenger class:\n", survival_rate_by_class)

# Get the row with the highest Fare
highest_fare_passenger = titanic.loc[titanic['Fare'].idxmax()]
print("\nPassenger with the highest fare:\n", highest_fare_passenger)

Survival rate by passenger class:
 Pclass
1    0.629630
2    0.472826
3    0.242363
Name: Survived, dtype: float64

Passenger with the highest fare:
 Survived                     1
Pclass                       1
Name          Ward, Miss. Anna
Sex                     female
Age                       35.0
SibSp                        0
Parch                        0
Ticket                PC 17755
Fare                  512.3292
Embarked                     C
FamilySize                   0
Name: 259, dtype: object


Conclusion:

*  Write a brief conclusion summarizing key findings and observations from the analysis.

📊 Conclusion: Titanic Dataset Analysis Summary

1.   Missing Values:
   *   Significant missing data was found in the 'Age', 'Cabin', and 'Embarked' columns.
   *   We handled missing values by imputing the median for 'Age' and the mode for 'Embarked'. The 'Cabin' column was dropped due to a high percentage of missing entries.
2.   Descriptive Statistics:
   *  The average passenger age was approximately 29 years.
   *  Fare values ranged widely, with a few high-paying passengers influencing the mean.
   *  Most passengers had no siblings/spouses or parents/children aboard.
3.   Feature Engineering:
   *  A new column 'FamilySize' was created, combining 'SibSp' and 'Parch', which can help in analyzing travel group dynamics.
4. Passenger Demographics:
   *  Unique passenger classes were 1, 2, and 3, with most passengers in 3rd class.
   *  Most passengers embarked from port 'S'.
5. Subset Analysis:
   *  A new subset of passengers over 30 years old was created and sorted by 'Fare', highlighting that older passengers often paid higher fares.
6. Survival Rates:
   *  1st class passengers had the highest survival rate (~63%), followed by 2nd (~47%) and 3rd (~24%). This suggests a strong class-based disparity in survival.
7. Highest Fare:
   *  The passenger who paid the highest fare (512.33) was in 1st class, indicating the luxury and exclusivity of accommodations at the top tier.